L1Trigger/L1THGCal/plugins/HGCalTriggerDigiProducer.cc

```diff
@@ -67,48 +69,70 @@ void HGCalTriggerDigiProducer::produce(edm::Event& e, const edm::EventSetup& es)
     fe_output( new l1t::HGCFETriggerDigiCollection );
   
   edm::Handle<HGCEEDigiCollection> ee_digis_h;
-  edm::Handle<HGCHEDigiCollection> fh_digis_h, bh_digis_h;
+  edm::Handle<HGCHEDigiCollection> fh_digis_h;
+  edm::Handle<HGCBHDigiCollection> bh_digis_h;
 
   e.getByToken(inputee_,ee_digis_h);
   e.getByToken(inputfh_,fh_digis_h);
-  //e.getByToken(inputbh_,bh_digis_h);
+  e.getByToken(inputbh_,bh_digis_h);
 
   const HGCEEDigiCollection& ee_digis = *ee_digis_h;
   const HGCHEDigiCollection& fh_digis = *fh_digis_h;
-  //const HGCHEDigiCollection& bh_digis = *bh_digis_h;
+  const HGCBHDigiCollection& bh_digis = *bh_digis_h;
 
   // First find modules containing hits and prepare list of hits for each module
   std::unordered_map<uint32_t, HGCEEDigiCollection> hit_modules_ee;
   for(const auto& eedata : ee_digis) {
     uint32_t module = triggerGeometry_->getModuleFromCell(eedata.id());
+    // Disconnect module = 0
```

Comment:
In the design of HGCAL some of the layers are not supposed to be used for the trigger. This is a first attempt to give the possibility to disconnect layers (and more generally modules). 
But I agree this implementation is not ideal. This part of the code should not have to know the internal implementation of the trigger geometry. A better way would be to define a method `isDisconnected(uint32_t)`  in the geometry class and define there what are the disconnected modules.
I am going to update the code in that direction.